dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/ResourceLink.java

```diff
@@ -95,50 +97,100 @@ public String toString() {
     public static class ResourceLinkBuilder {
 
         public final ResourceLink build(final HttpServletRequest request, final User user, final Contentlet contentlet) throws DotDataException, DotSecurityException {
-
-            if(!(contentlet.getContentType() instanceof FileAssetContentType)){
+            if(!contentlet.isFileAsset()){
                 throw new DotStateException(getLocalizedMessageOrDefault(user,"File-asset-contentlet-type-expected",
                         "Can only build Resource Links out of content with type `File Asset`.",getClass())
                 );
             }
+            return buildFileAsset(request, user, contentlet);
+        }
+        
+        
+        
+        public final ResourceLink build(final HttpServletRequest request, final User user, final Contentlet contentlet, final String velocityVarName) throws DotDataException, DotSecurityException {
+
 
+            File binary = Try.of(()->contentlet.getBinary(velocityVarName)).getOrNull();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'binary' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)